binderhub/static/js/pages/HomePage.jsx

```diff
@@ -0,0 +1,90 @@
+import { LinkGenerator } from "../components/LinkGenerator.jsx";
+import { BuilderLauncher } from "../components/BuilderLauncher.jsx";
+import { HowItWorks } from "../components/HowItWorks.jsx";
+import { useEffect, useState } from "react";
+import { FaviconUpdater } from "../components/FaviconUpdater.jsx";
+import { Spec, LaunchSpec } from "../spec.js";
+
+/**
+ * @typedef {object} HomePageProps
+ * @prop {import("../App.jsx").Provider[]} providers
+ * @prop {URL} publicBaseUrl
+ * @prop {URL} baseUrl
+ * @param {HomePageProps} props
+ */
+export function HomePage({ providers, publicBaseUrl, baseUrl }) {
+  const defaultProvider = providers[0];
+  const [selectedProvider, setSelectedProvider] = useState(defaultProvider);
+  const [repo, setRepo] = useState("");
+  const [ref, setRef] = useState("");
+  const [urlPath, setUrlPath] = useState("");
+  const [isLaunching, setIsLaunching] = useState(false);
+  const [spec, setSpec] = useState("");
+  const [progressState, setProgressState] = useState(null);
+
+  useEffect(() => {
+    const encodedRepo = selectedProvider.repo.urlEncode
+      ? encodeURIComponent(repo)
+      : repo;
+    let actualRef = "";
+    if (selectedProvider.ref.enabled) {
+      actualRef = ref !== "" ? ref : selectedProvider.ref.default;
+    }
+    setSpec(
+      new Spec(
+        `${selectedProvider.id}/${encodedRepo}/${actualRef}`,
+        new LaunchSpec(urlPath),
+      ),
+    );
+  }, [selectedProvider, repo, ref, urlPath]);
+
+  return (
+    <>
+      <div className="text-center">
```

Comment:
```suggestion
      <div className="text-center col-10 mx-auto">
```
The `App.jsx` change widened the box, this narrows just the text above the form.

![image](https://github.com/user-attachments/assets/e90b1524-5493-4861-b0a4-8543e8b87ccf)